### PR TITLE
Tools: add richenpower generator to build options

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: Tools/scripts/build_options.py [hunk 1]
```diff
@@ -81,6 +81,7 @@ def __init__(self,
     Feature('ICE', 'EFI_CURRAWONGECU', 'HAL_EFI_CURRAWONG_ECU_ENABLED', 'Enable EFI Currawong ECU', 0, 'EFI'),
     Feature('ICE', 'EFI_DRONECAN', 'HAL_EFI_DRONECAN_ENABLED', 'Enable EFI DroneCAN', 0, 'EFI'),
     Feature('ICE', 'GENERATOR', 'HAL_GENERATOR_ENABLED', 'Enable Generator', 0, None),
+    Feature('ICE', 'GENERATOR_RICHENPOWER', 'AP_GENERATOR_RICHENPOWER_ENABLED', 'Enable Richenpower Generator', 0, "GENERATOR"),  # noqa
```
Comment (rmackay9) [inline Tools/scripts/build_options.py:84]: The comment is a bit funny, "noqa".  I'm not sure we should leave that in.

Comment (peterbarker) [reply Tools/scripts/build_options.py:84]: > The comment is a bit funny, "noqa". I'm not sure we should leave that in.

There's a couple of dozen of them in this file.

It's short for "No Quality Assurance".  It's a directive to flake8 that there are known errors on this line and it should just move on with its nit-picking self.  In this case the line is too long.

**Review by amilcarlucas — APPROVED** [review]

**Review by Hwurzburg — COMMENTED** [review]
@peterbarker forgetting something???  extract_features.py?  I know I have recently...will correct

**Review by Hwurzburg — COMMENTED** [review]
nevermind....sorry about previous list here....forgot about the reg expressions picking up stuff

**Pull request merged**